src/test/java/com/google/firebase/auth/FirebaseAuthIT.java

```diff
@@ -705,6 +706,58 @@ public void testCustomTokenWithIAM() throws Exception {
     }
   }
 
+  @Test
+  public void testTenantAwareCustomToken() throws Exception {
+    // Create tenant to use.
+    TenantManager tenantManager = auth.getTenantManager();
+    Tenant.CreateRequest tenantCreateRequest =
+        new Tenant.CreateRequest().setDisplayName("DisplayName");
+    String tenantId = tenantManager.createTenant(tenantCreateRequest).getTenantId();
+
+    // Create and decode a token with a tenant-aware client.
+    TenantAwareFirebaseAuth tenantAwareAuth = auth.getTenantManager().getAuthForTenant(tenantId);
+    String customToken = tenantAwareAuth.createCustomTokenAsync("user1").get();
+    String idToken = signInWithCustomToken(customToken, tenantId);
+    FirebaseToken decoded = tenantAwareAuth.verifyIdTokenAsync(idToken).get();
+    assertEquals("user1", decoded.getUid());
+    assertEquals(tenantId, decoded.getTenantId());
+
+    // Delete tenant.
+    tenantManager.deleteTenantAsync(tenantId).get();
```

Comment:
Might want to push this into a finally block

Reply:
There's no existing try block to add the finally block to. Are you suggesting that I wrap the entire "Create and decode a token with a tenant-aware client" section with a try (and no catch)?

Reply:
Yes. You will see this pattern being used in a number of existing test cases. See `testCustomClaims()` for example.

Reply:
Okay sounds good! Done.